src/main.rs

```diff
@@ -134,6 +143,36 @@ impl Expr {
     }
 }
 
+impl Func {
+    fn eval(&self, arg: Value) -> Value {
+        match self {
+            Func::Expr(e) => e.eval(&[]),
+            Func::Fst => match arg {
+                Value::Pair(fst, _snd) => *fst,
+                _ => panic!("Fst: expected (_, _)"),
+            },
+            Func::Snd => match arg {
+                Value::Pair(_fst, snd) => *snd,
+                _ => panic!("Snd: expected (_, _)"),
+            },
+            Func::U16Be => match arg {
+                Value::Pair(fst, snd) => match (fst.as_ref(), snd.as_ref()) {
+                    (Value::U8(hi), Value::U8(lo)) => Value::U16(u16::from_be_bytes([*hi, *lo])),
+                    (_, _) => panic!("expected (U8, U8)"),
+                },
+                _ => panic!("U16Be: expected (_, _)"),
+            },
+            Func::Stream => match arg {
+                Value::Seq(vs) => {
+                    // FIXME could also condense nested sequences
+                    Value::Seq(vs.into_iter().filter(|v| *v != Value::Unit).collect())
```

Comment:
Note: Could use the `matches!` macro if you wanted to avoid needing `PartialEq`, but not a big deal.

Reply:
Oh that's a handy macro! Hacky but handy :laughing: